### PR TITLE
Move to Go 1.6

### PR DESCRIPTION
**Pull request by samuelkarp**
This PR moves the agent to Go 1.6 (finally!).  I've taken @euank's old Go 1.5 commit that he made before leaving and built on top of that.

r? @aaithal @juanrhenals @richardpen

### CODE REVIEW

File: agent/engine/docker_task_engine_test.go [hunk 1]
```diff
@@ -423,7 +423,7 @@ func TestSteadyStatePoll(t *testing.T) {
 	// Two steady state oks, one stop
 	gomock.InOrder(
 		client.EXPECT().DescribeContainer("containerId").Return(api.ContainerRunning, DockerContainerMetadata{DockerId: "containerId"}).Times(2),
-		client.EXPECT().DescribeContainer("containerId").Return(api.ContainerStopped, DockerContainerMetadata{DockerId: "containerId"}),
+		client.EXPECT().DescribeContainer("containerId").Return(api.ContainerStopped, DockerContainerMetadata{DockerId: "containerId"}).AnyTimes(),
```
Comment (aaithal) [inline agent/engine/docker_task_engine_test.go]: would it be better to add a TODO here to move this to `MinTimes(1)` ?

File: agent/engine/docker_task_engine_test.go [hunk 2]
```diff
@@ -579,7 +579,7 @@ func TestTaskTransitionWhenStopContainerTimesout(t *testing.T) {
 			// Since task is not in steady state, progressContainers causes
 			// another invocation of StopContainer. Return a timeout error
 			// for that as well
-			client.EXPECT().StopContainer(gomock.Any()).Return(containerStopTimeoutError),
+			client.EXPECT().StopContainer("containerId").Return(containerStopTimeoutError).AnyTimes(),
```
Comment (aaithal) [inline agent/engine/docker_task_engine_test.go]: same as above: would it be better to add a TODO here to move this to MinTimes(1) ?

**Comment by aaithal** [comment]
lgtm. just a couple of questions.

File: agent/engine/docker_task_engine_test.go (hunk 1 above)
Comment (samuelkarp) [inline agent/engine/docker_task_engine_test.go]: Sure, I can do that.  It's in a newer version of gomock, so we'll need to update that first before I can address the TODO.

Comment (aaithal) [inline agent/engine/docker_task_engine_test.go]: Yeah, I understand. let's add that to the TODO then. We can address that in a different PR.

Comment (samuelkarp) [inline agent/engine/docker_task_engine_test.go]: Done

**Comment by aaithal** [comment]
:shipit:

**Pull request merged**